### ISSUE TITLE
[FEATURE] Show training error in tensorboard in addition to validation error

### ISSUE THREAD

**Issue by fps** [sdatkinson/neural-amp-modeler#570]
While checking my own implementation of your "Standard"-Wavenet architecture I stumbled across something a little curious and I wanted to check whether the same thing happens in your implementation. For this I would need to see the average training error per epoch but I can't seem to find it in tensorboard.

I am not familiar with pytorch/lightning since I mostly use Julia/Flux for my experiments so the code is a little bit unpenetrable for me without taking lots of time to work through it. Can you give me hint, where to best add the training error to the lightning/tensorboard configuration?

Kind regards,
FPS

**Comment by sdatkinson** [comment]
Logging is handled by PyTorch Lightning; see https://lightning.ai/docs/pytorch/stable/extensions/logging.html#logging-from-a-lightningmodule for complete documentation

You want the `log_dict` method of the PyTorch Lightning module e.g. https://github.com/sdatkinson/neural-amp-modeler/blob/16274bacc4afb9f3551c9e15b647ca5a63aea735/nam/train/lightning_module.py#L345

Put something analogous in the training step method to get a per-iteration metric, or there are methods that are called at the start/end of a training epoch if you wanted to compute any metrics as a function of a whole epoch's iterations.

Hope that helps!

**Comment by fps** [comment]
Thanks! I hacked in a little bit of code to evaluate and log the ESR as well in the training step:

```python
    def training_step(self, batch, batch_idx):
        preds, targets, loss_dict = self._shared_step(batch)
        esr_loss = _esr(preds, targets)

        loss = 0.0
        for v in loss_dict.values():
            if v.weight is not None and v.weight > 0.0:
                loss = loss + v.weight * v.value


        self.log_dict(
                {
                    "train_loss": loss,
                    "train_loss_esr": esr_loss,
                    **{key: value.value for key, value in loss_dict.items()},
                }
        )

        return loss
```

 and it kind of confirms my suspicion:

![Image](https://github.com/user-attachments/assets/1a54a6ea-e830-4513-b693-c63f04f8d94b)

When reimplementing your Standard architecture and training it (purely by ESR) I was unable to train the network to validation losses that came close to the ones achieved with NAM. I did the usual thing: randomly split the data into training set and validation set. What was interesting though was that in some runs the validation error consistently was lower than the train error, which is weird in itself. I looked at the data (your example input/output.wav pair) and it occured to me that it was quite variable and especially contained some stretches of almost silence.. And if I got "lucky" I got a validation set that just had a lower ESR.

So I took a look at how you split the data and it seems you use the last 9 seconds or so as validation set. So I tried this as well and suddenly I got results like this:

```bash
 loss: trn: 5.771e-03, val: 6.175e-04 (min: 5.117e-04)
```

So the train loss is an order of magnitude higher than the validation loss.
